### PR TITLE
cdc: delay before sending data

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

I started testing the nano33 and found that the initial debug! messages ("Initialization...") were not showing up on my new mac with tockloader. This PR adds a 100 ms delay between when the CDC code detects a terminal is connected and when it actually starts sending data. This is inspired by https://michael.stapelberg.ch/posts/2021-04-27-linux-usb-virtual-serial-cdc-acm/.

I also added an additional check for receiving the `SET_CONTROL_LINE_STATE` message (per the blog post). So now the CDC start procedure looks like:

1. Start a 10 second timer. If this goes off we start dropping UART messages. Otherwise, we buffer and apply backpressure as necessary.
2. Wait for both `SET_CONTROL_LINE_STATE` and `SET_LINE_CODING` with baud rate 115200 to happen.
3. Then wait 100 ms.
4. Then start transmitting data.

### Testing Strategy

Verifying I can see the first debug prints on nano33ble on a new (but not M1 new) mac.

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: capsules/src/usb/cdc.rs [hunk 1]
```diff
@@ -416,29 +456,30 @@ impl<'a, U: hil::usb::UsbController<'a>, A: 'a + Alarm<'a>> hil::usb::Client<'a>
...
-                |line_coding| {
-                    // Check if we should switch our main state machine to
-                    // connecting meaning that the host is connecting to the virtual
-                    // serial port. We decide this based on if the host is
-                    // configuring the baud rate to what we expect.
-                    if self.state.get() == State::Enumerated && line_coding.baud_rate == 115200 {
-                        self.state.set(State::Connecting);
-                    }
+        match self.ctrl_state.get() {
+            CtrlState::SetLineCoding => {
```
Comment (alistair23) [inline capsules/src/usb/cdc.rs:460]: Why change this from an `if`?

Comment (bradjc) [reply capsules/src/usb/cdc.rs:460]: I guess that is left over from some things I tried that didn't end up being necessary. But if someone does want to handle data from a different ctrl message they likely want to make this change.

**Review by hudson-ayers — APPROVED** [review]

**Review by lschuermann — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]

**Comment by alevy** [comment]
Not to block on, but: I don't love about this that the 100ms delay "hack" is really for the console driver (when backed by USB) but is applied to all CDC instances (i.e. implemented at the CDC driver level). More ideal would be to apply the delay at _either_ the UART trait layer _or_ in the console driver (e.g., if there was a console driver specific for USB)

**Review by alevy — APPROVED** [review]
Approve with reservations in my comment above: a comment, maybe at the very top, specifying that this is a hack would be better. But also... whatever :)

**Pull request merged**